xapian-core/weight/tfidfweight.cc

```diff
@@ -149,6 +149,10 @@ TfIdfWeight::get_idfn(Xapian::doccount termfreq, char c) const
 	    // All documents are indexed by the term
 	    if (N == termfreq) return 0;
 	    return log((N - termfreq) / termfreq);
+        case 'f':
+            return (1.0/termfreq);
```

Comment:
This case doesn't use `N`, so line 144 above should avoid fetching it in this case too.